server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -364,4 +367,28 @@ private static void processMessageReceived(TransportRequest request, RequestHand
                                                TransportChannel transportChannel) throws Exception {
         requestHandler.processMessageReceived(request, transportChannel);
     }
+
+    private static Consumer<Runnable> wrap(Consumer<Runnable> runnableConsumer, DiscoveryNode node) {
+        return runnable -> runnableConsumer.accept(wrap(runnable, node));
+    }
+
+    private static Function<Runnable, Runnable> wrapper(DiscoveryNode node) {
+        return runnable -> wrap(runnable, node);
+    }
+
+    private static Runnable wrap(Runnable runnable, DiscoveryNode node) {
```

Comment:
Could this be `onNode(DiscoveryNode node, Runnable runnable)` - makes it a bit more readable at its call sites?